.konflux/scripts/render_snapshot.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python3
+
+import json
+import os
+import time
+
+
+def load_image_refs():
+    return json.loads(os.getenv("IMAGE_REFS", '[]'))
+
+
+def process_component(component, name_suffix):
+    if name_suffix != "":
+        name = f"{component['component']}-{name_suffix}"
+    else:
+        name = component["component"]
+    return {
+        "containerImage": component["ref"],
+        "name": name,
+        "source": {
+            "git": {
+                "revision": component["revision"],
+                "url": component["repository"]
+            }
+        }
+    }
+
+
+def construct_snapshot(
+    snapshot_name_prefix,
+    pipeline_run_name,
+    application,
+    components
+):
+    snapshot_name = f"{snapshot_name_prefix}-{int(time.time())}"
+    return {
+        "apiVersion": "appstudio.redhat.com/v1alpha1",
+        "kind": "Snapshot",
+        "metadata": {
+            "name": snapshot_name,
+            "labels": {
+                "appstudio.openshift.io/build-pipelinerun": pipeline_run_name
+            }
+        },
+        "spec": {
+            "application": application,
+            "components": components
+        }
+    }
+
+
+def determine_component_name_suffix(application):
+    return application.lstrip("acs-")
+
+
+if __name__ == '__main__':
+    application = os.getenv("APPLICATION", "")
+    pipeline_run_name = os.getenv("PIPELINE_RUN_NAME", "")
```

Comment:
Because the preferred solution is not to have a separate Python script (https://github.com/stackrox/stackrox/pull/13577#discussion_r1891808529), how about the following:

* We use `os.environ` and fail if the env var is missing
* We only access environment variables in the main function to read them and store their values in local variables.

Reply:
Could work but there are `args:` as a way to pass command-line arguments to the script. `args` does the same as passing options on the command line. Just feed it an array. See here https://github.com/stackrox/konflux-tasks/blob/b13ddf4691d9458fd1d9265c94a42cf4d9571068/tasks/fetch-scanner-v2-data-task.yaml#L51-L57